cli/tests/pcluster/api/controllers/test_cluster_operations_controller.py

```diff
@@ -798,13 +798,17 @@ def test_successful_request(
             mocker.patch(
                 "pcluster.models.cluster.Cluster.config_presigned_url", new_callable=mocker.PropertyMock
             ).return_value = "presigned-url"
+            config_mock = mocker.patch("pcluster.models.cluster.Cluster.config", new_callable=mocker.PropertyMock)
+            config_mock.return_value.scheduling.settings.scheduler_definition.metadata = metadata
+            config_mock.return_value.scheduling.scheduler = scheduler
         else:
             mocker.patch(
                 "pcluster.models.cluster.Cluster.config_presigned_url", new_callable=mocker.PropertyMock
             ).side_effect = ClusterActionError("failed")
-        config_mock = mocker.patch("pcluster.models.cluster.Cluster.config", new_callable=mocker.PropertyMock)
-        config_mock.return_value.scheduling.settings.scheduler_definition.metadata = metadata
-        config_mock.return_value.scheduling.scheduler = scheduler
+            mocker.patch(
```

Comment:
Could the check for `pcluster.models.cluster.Cluster.config_presigned_url` property fail with `ClusterActionError` before checking the `pcluster.models.cluster.Cluster.config` property? In which case this test might not cover the scenario where the scheduler metadata is missing?

Perhaps checking the logger for the message `Unable to retrieve scheduler metadata from cluster configuration.` can ensure this test is targetting the case where the Metadata is actually missing.